packages/meditrak-server/src/routes/GETSurveyScreenComponents/assertSurveyScreenComponentPermissions.js

```diff
@@ -34,17 +34,23 @@ export const createSurveyScreenComponentDBFilter = async (
   options,
   surveyId,
 ) => {
-  const dbConditions = { ...criteria };
+  let dbConditions = { ...criteria };
   const dbOptions = { ...options };
 
   if (surveyId) {
-    dbConditions.survey_id = surveyId;
+    dbConditions = {
+      'survey_screen.survey_id': surveyId,
+      ...criteria,
+    };
```

Comment:
This way you can still use a `const` in line 37 above, and it's a bit simpler

```suggestion
    dbConditions['survey_screen.survey_id'] = surveyId;
```

Unless if you expect that a `survey_screen.survey_id` may exist in `criteria`, and if it exists you want it to take precedence over the `surveyId` value